htdocs/modules/notifications/class/notification.php

```diff
@@ -401,9 +401,11 @@ public function triggerEvents(
         if (!is_array($events)) {
             $events = array($events);
         }
+        $status = true;
         foreach ($events as $event) {
-            $this->triggerEvent($category, $item_id, $event, $extra_tags, $user_list, $module_id, $omit_user_id);
+            $status = $status AND $this->triggerEvent($category, $item_id, $event, $extra_tags, $user_list, $module_id, $omit_user_id);
```

Comment:
This is suspect.

In general, try and avoid `AND` and `OR` logical operators. The precedence of these operators is usually confusing, and is totally different from the corresponding boolean operators, `&&` and `||`.

Example from the [PHP manual]( http://php.net/manual/en/language.operators.logical.php):

```
// The constant true is assigned to $h before the "and" operation occurs
// Acts like: (($h = true) and false)
$h = true and false;
```